client/src/components/Upload.vue

```diff
@@ -64,50 +103,72 @@ export default {
       }
       var uploaded = [];
       this.$emit("update:uploading", true);
-      await Promise.all(
-        this.pendingUploads.map(async pendingUpload => {
-          var { name, files, fps } = pendingUpload;
-          fps = parseInt(fps);
-          pendingUpload.uploading = true;
-          var { data: folder } = await this.girderRest.post(
-            "/folder",
-            `metadata=${JSON.stringify({
-              viame: true,
-              fps,
-              type: pendingUpload.type
-            })}`,
-            {
-              params: {
-                parentId: this.location._id,
-                name
-              }
-            }
-          );
-          var pending = files;
-          var results = [];
-          while (pending.length) {
-            results = results.concat(
-              await Promise.all(
-                pending.splice(0, 500).map(async file => {
-                  var uploader = new Upload(file, {
-                    $rest: this.girderRest,
-                    parent: folder
-                  });
-                  return await uploader.start();
-                })
-              )
-            );
-          }
-          uploaded.push({ folder, results, pipeline: pendingUpload.pipeline });
-          this.remove(pendingUpload);
-        })
-      );
+
+      // This is in a while loop to act like a Queue with it adding new items during upload
+      while (this.pendingUploads.length > 0) {
+        await this.uploadPending(this.pendingUploads[0], uploaded);
+      }
       this.$emit("update:uploading", false);
       this.$emit("uploaded", uploaded);
+    },
+    async uploadPending(pendingUpload, uploaded) {
+      var { name, files, fps } = pendingUpload;
+      fps = parseInt(fps);
+      pendingUpload.uploading = true;
+      var { data: folder } = await this.girderRest
+        .post(
+          "/folder",
+          `metadata=${JSON.stringify({
+            viame: true,
+            fps,
+            type: pendingUpload.type
+          })}`,
+          {
+            params: {
+              parentId: this.location._id,
+              name
+            }
+          }
+        )
+        .catch(error => {
+          if (
+            error.response &&
+            error.response.data &&
+            error.response.data.message
+          ) {
+            this.errorMessage = error.response.data.message;
+          } else {
+            this.errorMessage = error;
+          }
+          pendingUpload.uploading = false;
+          // Return empty object for the folder destructuring
+          return { data: null };
+        });
+
+      // function called after mixins upload finishes
+      const postUpload = data => {
+        uploaded.push({
+          folder,
+          results: data.results,
```

Comment:
Should there be a check here? When running locally, everytime I try to upload, it hands with an error saying:
```
TypeError: Cannot read property 'results' of undefined
```

Does this happen for anyone else?

Reply:
Not getting that error, but there should be a check there anyways.  Was you're girder web components updated in the process?

Reply:
I did not upgrade GWC. I'll upgrade and report back with the results.